Tentative Sparkfun Venus GPS support

makenai opened the pull request:
Started adding support for Sparkfun Venus GPS - could use some extra eyes to make sure I'm going in the right direction!

makenai committed:
Tentative Sparkfun Venus GPS support

makenai committed:
Tentative Sparkfun Venus GPS support

makenai commented:
Hey @dtex - do you think you could take a look at this and let me know if I'm going in the right direction?

The Venus GPS uses NMEA for standard GPS output, but a special binary format for input, so I added a few extra functions to handle those. I started working on some of the configuration functions, but it looks like there are some placeholders in the MT chip as well.

Related to those, I noticed this line:

``` javascript
        // Enforce maximum frequency of 10hz
        if (frequency < 10) {
          frequency = 10;
        }
```

Is that correct?

makenai committed:
Changed from references in the docs

makenai committed:
Changed from references in the docs

dtex commented:
Ah good catch. Yes, we have a maximum 10hz update frequency so we need to flip that comparison operator. It was ms instead of hz to begin with.

dtex commented:
This looks good to me. My only cooment is I wonder if it is possible to override the default listen() method by adding a new listen method to the Venus CHIP object. Then we could avoid adding those conditionals. If we start adding more chips that read in different ways into this one function, things could end up hard to manage.

makenai commented:
@dtex That's a good call. We don't want the other chipset listening for Skytraq specific things anyway. I'll look into splitting it out, and also getting some tests written soon.

makenai committed:
Separated Venus listener from NMEA listener. Added tests. Added retry logic since the board is a bit finicky.

makenai committed:
Separated Venus listener from NMEA listener. Added tests. Added retry logic since the board is a bit finicky.

dtex committed:
Merge remote-tracking branch 'makenai/addSparkFunGPS' into makenai-gps

dtex committed:
Breakout parsers into PARSERS object

Also make sure that chip specific commands overwrite the same commands
not he prototype.

dtex committed:
Breakout parsers into PARSERS object

Also make sure that chip specific commands overwrite the same commands
not he prototype.

dtex committed:
Merge remote-tracking branch 'origin/makenai-gps' into makenai-gps

makenai committed:
Merge pull request #3 from dtex/makenai-gps

Add PARSERS object
